shopping-cart/shopping-cart-scala/shopping-cart/src/main/scala/com/example/shoppingcart/impl/ShoppingCartReportProcessor.scala

```diff
@@ -0,0 +1,23 @@
+package com.example.shoppingcart.impl
+
+import com.lightbend.lagom.scaladsl.persistence.ReadSideProcessor
+import com.lightbend.lagom.scaladsl.persistence.slick.SlickReadSide
+
+
+class ShoppingCartReportProcessor(readSide: SlickReadSide,
```

Comment:
The EventProcessor and the Repository are separate classes and both are accessible on the cake. In other samples we've made the processor be an inner class of the repository ([example](https://github.com/lagom/online-auction-scala/blob/1.4.x/item-impl/src/main/scala/com/example/auction/item/impl/ItemRepository.scala#L116)). What's the benefit of this approach?

Reply:
In the linked example, the processor is not an inner class of repository. It's just a class defined on the same file. 

I find it a bad idea because we can't easily search for the class on GitHub or on an editor that doesn't have an index of the classpath.

Reply:
Fair enough.

About the link, I think I confused the java and the scala versions of `online-auction`  and probably we never made the move on the scala one.